pydicom/filereader.py

```diff
@@ -282,6 +294,12 @@ def data_element_generator(fp,
         if length != 0xFFFFFFFF:
             # don't defer loading of Specific Character Set value as it is
             # needed immediately to get the character encoding for other tags
+            if tag_set is not None and tag not in tag_set and tag != (
+                    0x08, 0x05):
+                # skip the tag if not in specific tags
+                fp.seek(fp_tell() + length)
+                continue
+
             if defer_size is not None and length > defer_size and tag != (
                     0x08, 0x05):
                 # Flag as deferred by setting value to None, and skip bytes
```

Comment:
This is Specific Character Set, one of the well known tags, that is read before all tags that can have an encoding - thus the low number. And yes, that's in the DICOM dictionary, but as I said, this specific one is well known, so nobody knowing DICOM has to look it up...